### PR TITLE
Upgrade Rocket from 0.3 to 0.4

### LINKED ISSUE #2: Upgrade Rocket 0.3 -> 0.4 🚀
Updating crates.io index
    Updating git repository `https://github.com/peterschwarz/hyper-sse.git`
error: failed to select a version for the requirement `ring = "^0.11.0"`
  candidate versions found which didn't match: 0.14.6, 0.14.5, 0.14.4, ...
  location searched: crates.io index
required by package `cookie v0.9.1`
    ... which is depended on by `rocket v0.3.0`
    ... which is depended on by `rocket_codegen v0.3.0`
    ... which is depended on by `cert-registry-rest-api v0.1.0 (/consensource/rest_api)`

### PR DESCRIPTION
**Pull request by trevormcdonald**
Hi all, these changes seem to, at least, make the type checker happy. This builds the rest_api stage successfully for me, but then errors at the processor stage when it tries to get the sawtooth-build-agent-os image. Should fix #2 . @adeebahmed let me know if anything else needs to be changed.

### CODE REVIEW

**Comment by pa3ng** [comment]
Hey @trevormcdonald, thanks for contributing! Your changes are running fine on my end 👍 But it looks like there are some conflicts with your branch that need to be resolved. Do you mind taking care of those and so we can merge this? Thanks!

**Comment by adeebahmed** [comment]
My comments are the same as @pa3ng 

Also, could you rebase your commits so only your contribution commits are part of the PR. (Drop my commits) 

It looks like you need to rebase your branch from master. 

Thanks for the contribution @trevormcdonald ! 😄

File: CONTRIBUTING [hunk 1]
```diff
@@ -3,7 +3,6 @@
 Contributions to this project are welcome.  If you'd like to contribute, raise an issue, or propose a new feature, please follow the procedures outlined below.
 
 ## Raising Issues
-A sample issue template is included in the .github directory for reference.
```
Comment (adeebahmed) [inline CONTRIBUTING]: This commit can be excluded when rebasing from master. I have an alias for this you may find useful to include in your .gitconfig

```
[alias]
        rbmaster = "!f() { echo "fetching upstream/master..." &&  git fetch upstream master &&  echo "rebasing on upstream/master..." && git rebase upstream/master; }; f"
        getpr = "!f() { git fetch upstream pull/"$1"/head:"$2" && git checkout "$2"; }; f"
        uncommit = reset --soft HEAD^
[rebase]
```

After adding that alias to your .gitconfig you can then do:
```
git stash
git rbmaster
git stash pop
```

Then add your commits and force push over this branch.

Comment (adeebahmed) [reply CONTRIBUTING]: Credit to @dplumb94 for the convenient alias 😄

Comment (trevormcdonald) [reply CONTRIBUTING]: Thanks for aliases! I cleaned up the commit history and resolved the conflicts, let me know if more changes are needed.

**Comment by pa3ng** [comment]
Hey @trevormcdonald, can you resolve these merge conflicts? You just need to resolve these conflicts in its own commit and then squash that merge resolve commit with the previous commit, i.e.  `Upgrade the rest_api to rocket 0.4 by updating the JSON handling and ... bfed93a` (it's a bit nit-picky but we try to consolidate commits per commit description, even w/ merge resolution commits). This should be the last update though 🙂

**Comment by trevormcdonald** [comment]
Sorry, I must be doing something wrong. I can resolve the conflicts, but when I try to squash the commit and force push to my branch, it shows more conflicts. I can get it to a state with one commit and conflicts, or two commits and no conflicts. Any ideas what I'm missing?

**Comment by pa3ng** [comment]
Hmm, hard to say what's happening without seeing it. I would have checked-out a working branch on my fork so that I could sink my fork's master with upstream master and then rebase my working branch with my fork's master as needed, and create a PR from the working branch. But there's a lot ways to go about it. 

Maybe you can try updating your fork's master to the upstream master, resolve the conflicts in a commit, then squash (or `fixup`) said commit to `b73a5c5`, then push again and see if that works? If that doesn't work, honestly I'm cool with having the two commits in history (@adeebahmed ).

**Comment by trevormcdonald** [comment]
@pa3ng  thanks for the suggestions. I tried pulling the upstream master, resolving the conflicts, and squashing the commits.

**Pull request closed without merging**